### PR TITLE
Some CLI additions and minor improvements

### PR DESCRIPTION
**Pull request by easterncalculus**
If interested, please consider testing these changes before merging as I have been unable to run locally; I think the issue is with my workstation.

### CODE REVIEW

File: src/game/game.ts [hunk 1]
```diff
@@ -145,7 +145,11 @@ export abstract class Game {
         const special = this.specials.draw()
         if (special) {
             player.addSpecial(special)
+<<<<<<< HEAD
```
Comment (North101) [inline src/game/game.ts]: One more to go :P

File: src/cli/player.ts [hunk 2]
```diff
@@ -234,49 +239,53 @@ export class PlayerCli extends Player {
...
-        } else if (report instanceof Reports.TeleporterPlayerReport) {
-            console.log(`${this.emoji} entered a teleporter`)
+            console.log(chalk.magenta(`${this.emoji} teleported to their starting space`))
         } else if (report instanceof Reports.TeleportPlayerReport) {
-            console.log(`${this.emoji} was teleported`)
+            console.log(chalk.bgMagenta.bold(`${this.emoji} was teleported!!`))
         } else if (report instanceof Reports.FoundPizzaPlayerReport) {
-            console.log(`${this.emoji} found a ${report.topping ? Topping[report.topping] + ' ' : ''}pizza!`)
+            if(report.topping){
+                console.log(chalk.bgBlue.bold(`${this.emoji} found a pizza! It's the ${report.topping ? Topping[report.topping] : 'None'} pizza!`))
```
[the remark this answers is not in the document]
Comment (North101) [reply src/cli/player.ts]: Also prevents the weird `It's the None pizza!` message (though it'll never happen)

File: src/cli/player.ts [hunk 3]
```diff
@@ -234,49 +239,53 @@ export class PlayerCli extends Player {
...
+            } else {
+                console.log(chalk.blue(`${this.emoji} found a pizza! But they already have the ${report.player.topping ? Topping[report.player.topping] : 'None'} pizza...`))
```
Comment (North101) [inline src/cli/player.ts]: If for some reason the player doesn't have a topping (which would be a programming mistake or maybe a special tile) the message would be `But they already have the None pizza...` Maybe the message should be `But they already have their hands full` or something similar.

**Comment by North101** [comment]
Thanks! I like the idea of the end game action and the colours (though not a fan of it being bg colours. Changing them to just font colours IMO looks better).

File: src/cli/player.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (easterncalculus) [reply src/cli/player.ts]: I did notice this, but decided to stick to the format used elsewhere for consistency. If you are happy with it I will shorten to just Topping[report.topping]

**Comment by easterncalculus** [comment]
> Thanks! I like the idea of the end game action and the colours (though not a fan of it being bg colours. Changing them to just font colours IMO looks better).

Just so you know my formatting rationale, I settled on:
- coloured text for differentiating
- bg colours for IMPORTANT INFO
as well as:
- red for attacks, as well as failures or mistakes (BG)
- green for success
- magenta for psychic powers, as well as getting teleported (BG)
- cyan for expansion content

but didn't get a good feel for them as I haven't been able to run locally. I felt like bg vs text colour would give a nice distinction between the two cases, but am open to changes.

File: src/game/game.ts (hunk 1 above)
Comment (easterncalculus) [reply src/game/game.ts]: Argh! I fixed this earlier but forgot to check it in.

File: src/game/actions.ts [hunk 4]
```diff
@@ -20,6 +20,12 @@ export class SkipAction extends Action {
     async resolve(game: Game) {}
 }
 
+export class EndGameAction extends Action {
+    async resolve(game: Game) {
+        game.turn = game.maxPlayerTurns
```
[the remark this answers is not in the document]
Comment (easterncalculus) [reply src/game/actions.ts]: fixed, good catch

File: src/cli/player.ts [hunk 5]
```diff
@@ -154,13 +157,13 @@ export class PlayerCli extends Player {
         return input.confirm
     }
 
-    async handleBackToStartSpecial(): Promise<Actions.AttackAction | Actions.MoveAction | Actions.SkipAction> {
+    async handleBackToStartSpecial(): Promise<Actions.AttackAction | Actions.MoveAction | Actions.SkipAction | Actions.EndGameAction> {
```
[the remark this answers is not in the document]
Comment (easterncalculus) [reply src/cli/player.ts]: removed

File: src/game/actions.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (North101) [reply src/game/actions.ts]: Actually on second thought: This should `throw new EndGameError()` where `EndGameError` inherits `GameOverError`

**Comment by North101** [comment]
> > Thanks! I like the idea of the end game action and the colours (though not a fan of it being bg colours. Changing them to just font colours IMO looks better).
> 
> Just so you know my formatting rationale, I settled on:
> 
> * coloured text for differentiating
> * bg colours for IMPORTANT INFO
>   as well as:
> * red for attacks, as well as failures or mistakes (BG)
> * green for success
> * magenta for psychic powers, as well as getting teleported (BG)
> * cyan for expansion content
> 
> but didn't get a good feel for them as I haven't been able to run locally. I felt like bg vs text colour would give a nice distinction between the two cases, but am open to changes.

On closer look, I don't mind the bgColour when its the full line, but I don't like it otherwise,

**Pull request merged**